x/stake/keeper/val_state_change.go

```diff
@@ -0,0 +1,287 @@
+
+// perform all the store operations for when a validator status becomes bonded
+func (k Keeper) bondValidator(ctx sdk.Context, validator types.Validator) types.Validator {
+
+	store := ctx.KVStore(k.storeKey)
+	pool := k.GetPool(ctx)
+
+	k.DeleteValidatorByPowerIndex(ctx, validator, pool)
+
+	validator.BondHeight = ctx.BlockHeight()
+
+	// XXX Are we OK with this? In order of power decreasing, but addresses break ties.
+	counter := k.GetIntraTxCounter(ctx)
+	validator.BondIntraTxCounter = counter
+	k.SetIntraTxCounter(ctx, counter+1)
+
+	// set the status
+	validator, pool = validator.UpdateStatus(pool, sdk.Bonded)
+	k.SetPool(ctx, pool)
+
+	// save the now bonded validator record to the three referenced stores
+	k.SetValidator(ctx, validator)
+	store.Set(GetValidatorsBondedIndexKey(validator.OperatorAddr), []byte{})
+
+	k.SetValidatorByPowerIndex(ctx, validator, pool)
+
+	// call the bond hook if present
+	if k.hooks != nil {
+		k.hooks.OnValidatorBonded(ctx, validator.ConsAddress())
+	}
+
+	return validator
+}
+
+// perform all the store operations for when a validator status begins unbonding
+func (k Keeper) beginUnbondingValidator(ctx sdk.Context, validator types.Validator) types.Validator {
+
+	store := ctx.KVStore(k.storeKey)
+	pool := k.GetPool(ctx)
+	params := k.GetParams(ctx)
+
+	k.DeleteValidatorByPowerIndex(ctx, validator, pool)
+
+	// sanity check
+	if validator.Status != sdk.Bonded {
+		panic(fmt.Sprintf("should not already be unbonded or unbonding, validator: %v\n", validator))
+	}
+
+	// set the status
+	validator, pool = validator.UpdateStatus(pool, sdk.Unbonding)
+	k.SetPool(ctx, pool)
+
+	validator.UnbondingMinTime = ctx.BlockHeader().Time.Add(params.UnbondingTime)
+	validator.UnbondingHeight = ctx.BlockHeader().Height
+
+	// save the now unbonded validator record
+	k.SetValidator(ctx, validator)
+
+	// also remove from the Bonded types.Validators Store
+	store.Delete(GetValidatorsBondedIndexKey(validator.OperatorAddr))
+
+	k.SetValidatorByPowerIndex(ctx, validator, pool)
+
+	// call the unbond hook if present
+	if k.hooks != nil {
+		k.hooks.OnValidatorBeginUnbonding(ctx, validator.ConsAddress())
+	}
+
+	return validator
+}
+
+// perform all the store operations for when a validator status becomes unbonded
+func (k Keeper) completeUnbondingValidator(ctx sdk.Context, validator types.Validator) types.Validator {
+	pool := k.GetPool(ctx)
+	validator, pool = validator.UpdateStatus(pool, sdk.Unbonded)
+	k.SetPool(ctx, pool)
+	k.SetValidator(ctx, validator)
+	return validator
+}
+
+// XXX need to figure out how to set a validator's BondIntraTxCounter - probably during delegation bonding?
+//     or keep track of tx for final bonding and set during endblock??? wish we could reduce this complexity
+
+// get the bond height and incremented intra-tx counter
+// nolint: unparam
+func (k Keeper) bondIncrement(ctx sdk.Context, isNewValidator bool, validator types.Validator) (
```

Comment:
where is this used? Also, it seems all the params are used?

Reply:
may be antiquated

Reply:
Thanks, antiquated, removed.